docs/spatialexample.rst

```diff
@@ -93,18 +93,18 @@ Model Class
 
             # Record results: reporting could actually be a component if we wanted. Or it can be
             # done in a log/report function in the relevant component (e.g., Transmission)
-            self.results.S[self.current_timestep, :] = np.array([
+            self.results.S[self.current_timestep, :] = [
                 np.sum(self.population.disease_state[self.population.node_id == i] == 0)
                 for i in range(self.nodes)
-            ])
-            self.results.I[self.current_timestep, :] = np.array([
+            ]
+            self.results.I[self.current_timestep, :] = [
                 np.sum(self.population.disease_state[self.population.node_id == i] == 1)
                 for i in range(self.nodes)
-            ])
-            self.results.R[self.current_timestep, :] = np.array([
+            ]
+            self.results.R[self.current_timestep, :] = [
                 np.sum(self.population.disease_state[self.population.node_id == i] == 2)
                 for i in range(self.nodes)
-            ])
+            ]
```

Comment:
This is not very performant (in case it is in the critical path). Using `np.add.at()` is ~10x faster.

```python
np.add.at(self.results.S[self.current_timestep], node_id[state == 0], 1)
np.add.at(self.results.I[self.current_timestep], node_id[state == 1], 1)
np.add.at(self.results.R[self.current_timestep], node_id[state == 2], 1)
```

Admittedly, `np.add.at()` is not necessarily well known or as easily grokked.